pkgs/development/interpreters/python/cpython/2.7/default.nix

```diff
@@ -118,7 +118,7 @@ let
         ] ++ optionals x11Support [ tcl tk xlibsWrapper libX11 ]
     )
     ++ optional zlibSupport zlib
-    ++ optional stdenv.isDarwin CF;
+    ++ optionals stdenv.isDarwin [ coreutils CF ];
```

Comment:
Yeah, that's how Nix does all runtime dependencies. `buildInputs` merely affects which hooks are run at build time, and is not persisted anywhere on disk. `propagatedBuildInputs` boils down to writing a file in the derivation output that encodes literal paths to the dependencies.